### PR TITLE
support multiple ConfigServers

### PR DESCRIPTION
**Pull request by zetaab**
Instead of supporting one boostrap ConfigServer, I modified it to understand multiple ConfigServers. The logic is that it will try to find the working one from the all available configservers. It seems that Kops-controller will answer only from one master

### CODE REVIEW

File: upup/pkg/fi/nodeup/command.go
Comment (zetaab) [inline upup/pkg/fi/nodeup/command.go:166]: this is little bit dirty, but all configservers do have same CA Certificate

File: pkg/apis/nodeup/config.go
Comment (hakman) [inline pkg/apis/nodeup/config.go:119]: Not quite sure if `ConfigServers` or adding `Servers` to `ConfigServerOptions`. `Server = Servers[0]` could be kept around for backward compatibility.

File: upup/pkg/fi/nodeup/command.go
Comment (hakman) [inline upup/pkg/fi/nodeup/command.go:765]: I don't think an extra function is needed for now.

File: upup/pkg/fi/nodeup/command.go
Comment (hakman) [inline upup/pkg/fi/nodeup/command.go:775]: Would be simpler to just use `"go.uber.org/multierr"`:
```suggestion
	var merr error
	for _, configServer := range configServers {
		u, loopErr := url.Parse(configServer.Server)
		if loopErr != nil {
			merr = multierr.Append(merr, fmt.Errrorf("unable to parse configuration server url %q: %w", configServer.Server, loopErr))
```

File: upup/pkg/fi/nodeup/command.go
Comment (hakman) [inline upup/pkg/fi/nodeup/command.go:789]:
```suggestion
			merr = multierr.Append(merr, err)
```

**Review by hakman — CHANGES_REQUESTED** [review]
I am generally ok with the direction. Leaving the final review to @justinsb or @johngmyers.
/assign @johngmyers 
/assign @justinsb

**Comment by johngmyers** [comment]
I'll have to research why it only answers from one instance.

**Comment by johngmyers** [comment]
Okay, this was introduced in #12997 

The server should not be put under control of the manager, since that subjects it to leader election.

**Comment by johngmyers** [comment]
@zetaab I'd appreciate if you could try #15002.

**Comment by hakman** [comment]
I think without putting all 3 servers (when available) there is less redundancy. We already send all 3 IPs for the bootstrap check.
Also, seems this is the direction in which @justinsb was going:
https://github.com/kubernetes/kops/pull/14501#discussion_r1020973316
https://github.com/kubernetes/kops/pull/14769#discussion_r1049717186

File: pkg/apis/nodeup/config.go
Comment (johngmyers) [reply pkg/apis/nodeup/config.go:119]: It would be better to have `Servers` in `ConfigServerOptions`. The servers aren't going to have different CAs.

File: upup/pkg/fi/nodeup/command.go
Comment (johngmyers) [inline upup/pkg/fi/nodeup/command.go:726]:
```suggestion
// getNodeConfigFromServers queries kops-controllers for our node's configuration.
```

File: upup/pkg/fi/nodeup/command.go
Comment (johngmyers) [reply upup/pkg/fi/nodeup/command.go:765]: Concur

File: upup/pkg/fi/nodeup/command.go
Comment (johngmyers) [inline upup/pkg/fi/nodeup/command.go:775]:
```suggestion
			return nil, fmt.Errorf("parsing configuration server url %q: %w", configServer.Server, loopErr)
```

**Review by johngmyers — CHANGES_REQUESTED** [review]
Next time, when both the change and the changes to golden outputs are nontrivial please put the golden output changes in a separate commit.

File: upup/pkg/fi/nodeup/command.go
Comment (johngmyers) [inline upup/pkg/fi/nodeup/command.go:167]: `nodeConfig` can only be non-nil if this condition is true, so there's no need to test it.

File: upup/pkg/fi/nodeup/command.go
Comment (johngmyers) [inline upup/pkg/fi/nodeup/command.go:775]: No need to move this inside the loop.

**Review by johngmyers — APPROVED** [review]
/hold to give opportunity to fix nits

File: upup/pkg/fi/nodeup/command.go
Comment (johngmyers) [inline upup/pkg/fi/nodeup/command.go:116]: You need to keep the previous nil check or it will NPE if no ConfigServer

File: upup/pkg/fi/nodeup/command.go
Comment (hakman) [inline upup/pkg/fi/nodeup/command.go:776]:
```suggestion
		u, err := url.Parse(server)
		if err != nil {
			merr = multierr.Append(merr, fmt.Errorf("unable to parse configuration server url %q: %w", server, loopErr))
			continue			
```

File: upup/pkg/fi/nodeup/command.go
Comment (hakman) [inline upup/pkg/fi/nodeup/command.go:790]:
```suggestion
		err = client.Query(ctx, &request, &resp)
		if err != nil {
			merr = multierr.Append(merr, err)
			continue
		}
		return &resp, nil
```

**Review by hakman — CHANGES_REQUESTED** [review]

**Comment by hakman** [comment]
Please restage/squash the commits before merging. Will be easier to cherrypick.
/lgtm

**Comment by hakman** [comment]
I meant to keep all code in one PR and the golden outputs, as @johngmyers said.
When we cherry-pick, most likely the code will not have any changes, but the expected will be regenerated. It's much easier later to track if a commit was ported to a branch or not.

**Review by hakman — APPROVED** [review]

**Pull request merged**
